lints/arbitrary_cpi/src/lib.rs

```diff
@@ -60,6 +60,24 @@ dylint_linting::declare_late_lint! {
     /// };
     /// invoke(&ix, accounts.clone());
     /// ```
+    /// 
+    /// **How the lint is implemented:**
+    /// 
+    /// - For every function containing calls to `solana_program::program::invoke`
+    /// - find the definition of `Instruction` argument passed to `invoke`; first argument
+    /// - If the `Instruction` argument is result of a function call
+    ///     - If the function is whitelisted, do not report; only functions defined in `spl_token::instruction` are whitelisted.
+    ///     - Else report the call to `invoke` as vulnerable
+    /// - Else if the `Instruction` is initialized in the function itself
+    ///     - find the assign statement assigning to the `program_id` field, assigning to field at `0`th index
+    ///     - find all the aliases of `program_id`. Use the rhs of the assignment as initial alias and look for
+    ///         all assignments assigning to the locals recursively.
+    ///     - Check if `program_id` is compared using any of aliases.
+    ///         - look for calls to `core::cmp::PartialEq{ne, eq}` where one of arg is moved from an alias.
+    ///         - If one of the arg accesses `program_id`, check if the basic block containing the comparison
+    ///         dominates the basic block containing call to `invoke` ensuring the `program_id` is checked in all execution
+    ///         paths.
+    ///         - If basic block does not dominate or there is no such comparison report the call to `invoke`
```

Comment:
Is this sentence incomplete?

Reply:
No, the sentence is complete. Is anything not clear?